src/GoogleFit/FitOnFhir.GoogleFit/Services/GoogleFitImportService.cs

```diff
@@ -97,6 +109,11 @@ public class GoogleFitImportService : ParallelTaskWorker<GoogleFitImportOptions>
                             await _eventHubProducerClient.SendAsync(eventBatch, cancellationToken);
                         }
                         while (pageToken != null);
+
+                        // Update the GoogleFitUser timestamp for this data source
+                        user.SyncTimes.Remove(dataStreamId);
+                        user.SyncTimes.Add(dataStreamId, currentTime);
```

Comment:
You can just do `user.SyncTimes[dataStreamId] = currentTime;` to update the value in the dictionary.

Reply:
Good point.